lib/jsdom/living/nodes/GlobalEventHandlers-impl.js

```diff
@@ -1,6 +1,202 @@
+
+    } else {
+      returnValue = callback.call(event.currentTarget, event);
+    }
+
+    if (eventName === "mouseover" || (eventName === "error" && ErrorEvent.isImpl(event))) {
+      if (returnValue) {
+        event._canceledFlag = true;
+      }
+    // TODO: before unload
+    } else if (!returnValue) {
+      event._canceledFlag = true;
+    }
+  }, false);
+}
+
+function createEventMethod(obj, event) {
+  Object.defineProperty(obj, "on" + event, {
+    get() {
+      if (inheritedAttributes.has(event) && (this.tagName === "BODY" || this.tagName === "FRAMESET")) {
+        return this.ownerDocument.defaultView["on" + event] === undefined ? null : this.ownerDocument.defaultView["on" + event];
+      }
+
+      const value = this._eventHandlers[event];
+      if (!value) {
+        return null;
+      }
+
+      if (value.body !== undefined) {
+        let element;
+        let formOwner = null;
+        let document;
+        let window;
+        if (this.constructor.name === "Window") {
+          element = null;
+          window = this;
+          document = idlUtils.implForWrapper(this.document);
+        } else {
+          element = this;
+          document = element.ownerDocument;
+          window = document.defaultView;
+        }
+        const body = value.body;
+        // TODO: location
+        if (element !== null) {
+          formOwner = element.form || null;
+        }
+
+        try {
+          // eslint-disable-next-line no-new-func
+          Function(body); // properly error out on syntax errors
+        } catch (e) {
+          // TODO: Report the error
+          return null;
+        }
+
+        let fn;
+        const Constructor = vm.isContext(document._global) ? document.defaultView._globalProxy.Function : Function;
+        if (event === "error" && element === null) {
+          // eslint-disable-next-line no-new-func
+          fn = new Constructor(`return function onerror(event, source, lineno, colno, error) {
+  ${body}
+};`)(document);
+        } else {
+          const argNames = [];
+          const args = [];
+          if (element !== null) {
+            argNames.push("document");
+            args.push(document);
+          }
+          if (formOwner !== null) {
+            argNames.push("formOwner");
+            args.push(formOwner);
+          }
+          if (element !== null) {
+            argNames.push("element");
+            args.push(element);
+          }
+          let wrapperBody = `
+return function on${event}(event) {
+  ${body}
+};`;
+          for (let i = argNames.length - 1; i >= 0; --i) {
+            wrapperBody = `with (${argNames[i]}) { ${wrapperBody} }`;
+          }
+          argNames.push(wrapperBody);
+          fn = Constructor(...argNames)(...args);
+        }
+        this._eventHandlers[event] = fn;
+      }
+      return this._eventHandlers[event];
+    },
+    set(val) {
+      if (!(typeof val === "function")) {
```

Comment:
As per discussion, since we need to mix in the impl directly for Window, I guess we need this. In which case it needs a comment explaining this.